pkg/cloudprovider/gce/gce.go

```diff
@@ -379,11 +383,53 @@ func (gce *GCECloud) CreateTCPLoadBalancer(name, region string, externalIP net.I
 		return nil, err
 	}
 
+	allowedPorts := make([]string, len(ports))
+	for ix := range ports {
+		allowedPorts[ix] = strconv.Itoa(ports[ix].Port)
+	}
+
+	hostTag := gce.computeHostTag(hosts[0])
+
+	firewall := &compute.Firewall{
+		Name:         makeFirewallName(name),
+		Description:  fmt.Sprintf("KubernetesAutoGenerated_OnlyAllowTrafficForDestinationIP_%s", fwd.IPAddress),
+		Network:      gce.gceNetworkName(),
+		SourceRanges: []string{"0.0.0.0/0"},
+		TargetTags:   []string{hostTag},
+		Allowed: []*compute.FirewallAllowed{
+			{
+				IPProtocol: "tcp",
+				Ports:      allowedPorts,
+			},
+		},
+	}
+	if op, err = gce.service.Firewalls.Insert(gce.projectID, firewall).Do(); err != nil {
+		return nil, err
+	}
+	if op.Status == "DONE" && op.Error != nil {
+		// An error occurred
+		if op.HttpErrorStatusCode == http.StatusConflict {
+			glog.V(4).Infof("Detected an already existing firewall: %s", firewall.Name)
+		} else {
+			var msg string
+			if len(op.Error.Errors) > 0 {
+				msg = op.Error.Errors[0].Message
+			}
+			return nil, fmt.Errorf("GCE Operation error: %s (%d %s)", msg, op.HttpErrorStatusCode, op.HttpErrorMessage)
+		}
+	}
+
 	status := &api.LoadBalancerStatus{}
 	status.Ingress = []api.LoadBalancerIngress{{IP: fwd.IPAddress}}
 	return status, nil
 }
 
+// This is kind of hacky, but the managed instance group adds 4 random chars and a hyphen
+// to the base name.
+func (gce *GCECloud) computeHostTag(host string) string {
+	return host[:len(host)-5]
```

Comment:
Is that part of a real API guarantee of how many characters that will be added? We could reuse the logic from [our config file](https://github.com/GoogleCloudPlatform/kubernetes/blob/ffb846a284fd0fd125e331fb01b26a8d4650ea9c/cluster/gce/config-default.sh#L41), perhaps?

It also adds an expectation that users will actually add tags to the VMs if they try assembling a cluster themselves, although I'm less sympathetic to this sort of breakage.